pdf_parsing/parser.py

```diff
@@ -2,102 +2,152 @@
+        lines = text.replace("\t", " ").split("\n")
+
+        quote_lines = []
+        explanation_lines = []
+        quote_found = False
+        quote_source_found = False
+
+        explanation_first_letter, date, title, next_lines = (
+            self.__parse_letter_date_title(lines)
+        )
+
+        for index, line in enumerate(next_lines):
+            line = line.strip()
+
+            if not quote_found:
+                quote_lines.append(line)
+                quote_found = self.__is_end_of_quote(
+                    line, next_lines[index + 1].strip()
+                )
                 continue
-            if page.number > end:
-                break
-            if page.number in breaks:
+
+            if not quote_source_found and (
+                line[0] == START_SOURCE_SYMBOL or line[0:5].isupper()
+            ):
+                quote_source = START_SOURCE_SYMBOL
+                # This is required to handle cases where the START_SOURCE_SYMBOL is missing
+                quote_source += line[1:] if (line[0] == START_SOURCE_SYMBOL) else line
+                quote_source = quote_source.strip()
+                quote_source_found = True
                 continue
 
-            date, title, quote, source, explanation = parse_page_text(page.getText())
-            entries[date] = {
-                "title": title,
-                "quote": quote,
-                "quote_source": source,
-                "explanation": explanation
-            }
+            explanation_lines.append(line)
 
-    return entries
+        date = date[:-2]
+        quote = self.__apply_quotes(" ".join(quote_lines).strip())
+        explanation = (explanation_first_letter + " ".join(explanation_lines)).strip()
 
+        return date, title, quote, quote_source, explanation
 
-if __name__ == "__main__":
+    # Extracts the first letter of explanation, date and title from the first lines of the page
+    def __parse_letter_date_title(self, lines):
+        index = 0
+        explanation_first_letter = ""
+
+        # Required to cover cases where the first letter is empty or the first letter is prefixed with quote.
+        if len(lines[index]) == 1 or len(lines[index]) == 2:
+            explanation_first_letter = lines[index].strip()
+            index += 1
+
+        date = lines[index].strip()
+        index += 1
 
-    # These are specific to Ishan's copy of the Daily Stoic
-    start_page = 14
-    end_page = 392
-    skip = [45, 75, 107, 138, 139, 171, 202, 234, 266, 267, 298, 330, 361]
+        # Required to cover cases where the title is split into multiple lines.
+        title_lines = []
+        while lines[index].isupper():
+            title_lines.append(lines[index])
+            index += 1
+        title = "".join(title_lines).strip()
 
-    data = extract_text("The Daily Stoic.pdf", start=start_page, end=end_page, breaks=skip)
+        return explanation_first_letter, date, title, lines[index:]
```

Comment:
same here, a named tuple as a way of grouping variables could make the return simpler here